cli/pa_analyze_command.go

```diff
@@ -0,0 +1,1155 @@
+	allExamples   bool
+	checks        []struct {
+		checkName string
+		checkFunc func(reader *archive.Reader) (paCheckOutcome, error)
+	}
+}
+
+type paCheckOutcome int
+
+func (s paCheckOutcome) badge() string {
+	switch s {
+	case Pass:
+		return "✅ PASS"
+	case Fail:
+		return "❌ FAIL"
+	case SomeIssues:
+		return "⚠️ WARN"
+	case Skipped:
+		return "◻️ SKIP"
+	default:
+		panic(s)
+	}
+}
+
+const (
+	Skipped paCheckOutcome = iota
+	Pass
+	Fail
+	SomeIssues
+)
+
+func configurePaAnalyzeCommand(srv *fisk.CmdClause) {
+	c := &paAnalyzeCmd{}
+	c.checks = []struct {
+		checkName string
+		checkFunc func(reader *archive.Reader) (paCheckOutcome, error)
+	}{
+		{
+			"Server health",
+			c.checkServerHealth,
+		},
+		{
+			"Uniform server version",
+			c.checkServerVersions,
+		},
+		{
+			"Slow consumers",
+			c.checkSlowConsumers,
+		},
+		{
+			"Cluster memory usage",
+			c.checkClusterMemoryUsageOutliers,
+		},
+		{
+			"Lagging stream replicas",
+			c.checkLaggingStreamReplicas,
+		},
+		{
+			"CPU usage",
+			c.checkCpuUsage,
+		},
+		{
+			"High subject cardinality streams",
+			c.checkHighSubjectCardinalityStreams,
+		},
+		{
+			"High number of HA assets",
+			c.checkHighHAAssetCardinality,
+		},
+		{
+			"Reserved resources limit",
+			c.checkResourceLimits,
+		},
+		{
+			"Account limits",
+			c.checkAccountLimits,
+		},
+		{
+			"Stream limits",
+			c.checkStreamLimits,
+		},
+		{
+			"Meta cluster offline replicas",
+			c.checkMetaClusterOfflineReplicas,
+		},
+		{
+			"Meta cluster leader",
+			c.checkMetaClusterLeader,
+		},
+		{
+			"Configured gateways",
+			c.checkConfiguredGateways,
+		},
+	}
+
+	analyze := srv.Command("analyze", "perform checks against an archive generated by the Gather subcommand").Action(c.analyze)
+	analyze.Arg("archive-path", "path to input archive to analyze").Required().StringVar(&c.archivePath)
+	analyze.Flag("examples", "Maximum number of example issues to display per check").Default("5").UintVar(&c.examplesLimit)
+	analyze.Flag("all-examples", "Display all issues detected by each check").UnNegatableBoolVar(&c.allExamples)
+	// Hidden flags
+	analyze.Flag("very-verbose", "Print a lot of intermediate detailed during analysis").Hidden().BoolVar(&c.veryVerbose)
+
+}
+
+func (cmd *paAnalyzeCmd) analyze(_ *fisk.ParseContext) error {
+	// Configure based on options
+	if cmd.allExamples {
+		cmd.examplesLimit = 0
+	}
+
+	// Open archive
+	ar, err := archive.NewReader(cmd.archivePath)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := ar.Close()
+		if err != nil {
+			fmt.Printf("Failed to close archive reader: %s\n", err)
+		}
+	}()
+
+	// Run all configured checks
+	for _, check := range cmd.checks {
+		fmt.Printf("\n")
+		cmd.logDebug("Running check: %s", check.checkName)
+		outcome, err := check.checkFunc(ar)
+		if err != nil {
+			return fmt.Errorf("check '%s' error: %w", check.checkName, err)
+		}
+		fmt.Printf("%s - %s\n\n", outcome.badge(), check.checkName)
+	}
+
+	return nil
+}
+
```

Comment:
All of the checks below could be moved to a different package if that's your preference